src/algorithms/Algorithm.cpp

```diff
@@ -142,9 +142,9 @@ void Algorithm::write_clustering_to_file(const std::string &filename) const {
 }
 
 //! \param filename Name of file to write to
-void Algorithm::write_density_to_file(const std::string &filename) const {
-  if (!density_was_computed) {
-    std::cerr << "Error: cannot write density to file; eval_density() "
+void Algorithm::write_lpdf_to_file(const std::string &filename) const {
+  if (!lpdf_was_computed) {
```

Comment:
I would move this function to a I/O utils file or something like that, I think the algorithm should just return the lpdf evaluated on a grid!